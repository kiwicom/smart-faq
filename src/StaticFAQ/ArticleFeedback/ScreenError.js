// @flow

import * as React from 'react';
import css from 'styled-jsx/css';
import { Alert } from '@kiwicom/orbit-components';
import { Close } from '@kiwicom/orbit-components/lib/icons';

import screensList from './screensList';
import { Box } from '../../common';

type Props = {|
  changeScreen: (nextScreen: string) => void,
|};

const style = css`
  div.thankYou {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 251px;
    padding-left: 14px;
    padding-right: 14px;
  }
  div.closeIcon {
    position: absolute;
    top: 8px;
    right: 8px;
    cursor: pointer;
  }
  div.systemMessage {
    margin-bottom: 16px;
  }
`;
const ScreenError = (props: Props) => {
  return (
    <Box border="none" borderRadius="4px" backgroundColor="#f5f7f9">
      <div className="thankYou">
        <div
          className="closeIcon"
          onClick={() => props.changeScreen(screensList.INITIAL)}
          onKeyUp={null}
          tabIndex="-1"
          role="button"
        >
          <Close customColor="#bac7d5" size="small" />
        </div>
        <div className="systemMessage">
          <Alert type="warning" icon>
            Our bad. We weren&apos;t able to send your feedback. But, follow the
            link below to drop us a line.
          </Alert>
        </div>
        <style jsx>{style}</style>
      </div>
    </Box>
  );
};

export default ScreenError;
