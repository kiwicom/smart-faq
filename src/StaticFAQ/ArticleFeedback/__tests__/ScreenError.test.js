// @flow

import * as React from 'react';
import { render } from 'enzyme';

import ScreenError from '../ScreenError';

const ch = (sc: string) => {}; //eslint-disable-line

describe('ScreenError', () => {
  const component = render(<ScreenError changeScreen={ch} />);
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
