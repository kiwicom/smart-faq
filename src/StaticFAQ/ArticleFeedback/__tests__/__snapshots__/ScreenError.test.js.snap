// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenError should match snapshot 1`] = `
<Box
  backgroundColor="#f5f7f9"
  border="none"
  borderRadius="4px"
>
  <div
    className="jsx-1340655337 thankYou"
  >
    <div
      className="jsx-1340655337 closeIcon"
      onClick={[Function]}
      onKeyUp={null}
      role="button"
      tabIndex="-1"
    >
      <Close
        customColor="#bac7d5"
        size="small"
      />
    </div>
    <div
      className="jsx-1340655337 systemMessage"
    >
      <SystemMessage
        Icon={[Function]}
        title="Feedback form failed"
        type="warning"
      >
        Our bad. We weren't able to send your feedback. But, follow the link below to drop us a line.
      </SystemMessage>
    </div>
    <JSXStyle
      css="div.thankYou.jsx-1340655337{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:251px;padding-left:14px;padding-right:14px;}div.closeIcon.jsx-1340655337{position:absolute;top:8px;right:8px;cursor:pointer;}div.systemMessage.jsx-1340655337{margin-bottom:16px;}"
      styleId="1340655337"
    />
  </div>
</Box>
`;
